src/types.ts

```diff
@@ -18,3 +20,10 @@ export type SerializedInput<Node = unknown, LinkProps = unknown> = {
 };
 
 export type NoInfer<T> = [T][T extends any ? 0 : never];
+
+export type NextWeightFnParams<Node = unknown, LinkProps = unknown> = {
```

Comment:
I think we should also add the "path" as parameter, that would allow to get the total number of hops for the cases when the calculation is based on the distance from the source

Reply:
This is done.